<launch>
  <arg name="num_target" default="1"/>
  <arg name="num_robot" default="1"/>
  <arg name="verbal_flag" default="false"/>
  <arg name="num_motion_primitive" default="8"/>
  <arg name="time_interval" default="8"/>
  <arg name="robot_time" default="1"/>
  <arg name="robot_trans_speed" default="0.5"/>
  <arg name="robot_ang_speed" default="0.5"/>
  <arg name="sensing_range" default="5"/>
  <arg name="comm_range" default="5"/>

  <node type="get_target_odom_journal_node" pkg="max_min_lp_simulation" name="get_target_odom_journal_node" output="screen"/>

  <group ns="robot_0">
    <arg name="robot_id" default="0"/>
    <arg name="robot_name" default="robot_0"/>
    <node type="max_min_lp_simulation_robot_journal" pkg="max_min_lp_simulation" name="max_min_lp_simulation_robot_journal" output="screen">
      <param name="num_robot" type="int" value="$(arg num_robot)"/>
      <param name="num_target" type="int" value="$(arg num_target)"/>
      <param name="robot_id" type="int" value="$(arg robot_id)"/>
      <param name="robot_name" type="str" value="$(arg robot_name)"/>
      <param name="verbal_flag" type="bool" value="false"/>
      <param name="num_motion_primitive" type="int" value="$(arg num_motion_primitive)"/>
      <param name="time_interval" type="int" value="$(arg time_interval)"/>
      <param name="robot_time" type="int" value="$(arg robot_time)"/>
      <param name="robot_trans_speed" type="double" value="$(arg robot_trans_speed)"/>
      <param name="robot_ang_speed" type="double" value="$(arg robot_ang_speed)"/>
      <param name="sensing_range" type="double" value="$(arg sensing_range)"/>
      <param name="comm_range" type="double" value="$(arg comm_range)"/>
    </node>

    <node type="apply_motion_primitive_journal" pkg="max_min_lp_simulation" name="apply_motion_primitive_journal" output="screen">
      <param name="num_target" type="int" value="$(arg num_target)"/>
      <param name="num_robot" type="int" value="$(arg num_robot)"/>
      <param name="robot_id" type="int" value="$(arg robot_id)"/>
      <param name="robot_name" type="str" value="$(arg robot_name)"/>
    </node>

    <node type="get_odom_node" pkg="max_min_lp_simulation" name="get_odom_node" output="screen">
      <param name="num_target" type="int" value="$(arg num_target)"/>
      <param name="num_robot" type="int" value="$(arg num_robot)"/>
      <param name="robot_id" type="int" value="$(arg robot_id)"/>
      <param name="robot_name" type="str" value="$(arg robot_name)"/>
    </node>
  </group>
</launch>
